### ISSUE TITLE
Cannot find the xml direction in set_ur5_joint

### ISSUE THREAD

**Issue by Gloriabhsfer** [roboticsleeds/mujoco_ur5_model#1]
Hi:
Thanks for your good model.
I am trying modify robot joint through set_ur5_joint. In that program, it will pop out a message that I need to type the directory where xml file I want to modify.
So I open a empty xml file and paste 

> <body name="shoulder_link" pos="0.28 0 0.545159" quat="0.681998 0 0 -0.731354">
    <inertial pos="0 0 0" mass="3.7" diaginertia="0.0102675 0.0102675 0.00666" />
    <geom type="mesh" rgba="0.7 0.7 0.7 1" mesh="shoulder" />
    <body name="upper_arm_link" pos="0 0.13585 0" quat="0.707107 0 0.707107 0">
        <inertial pos="0 0 0.28" mass="8.393" diaginertia="0.226891 0.226891 0.0151074" />
        <geom type="mesh" rgba="0.7 0.7 0.7 1" mesh="upperarm" />
        <body name="forearm_link" pos="0 -0.1197 0.425">
            <inertial pos="0 0 0.25" mass="2.275" diaginertia="0.0494433 0.0494433 0.004095" />
            <geom type="mesh" rgba="0.7 0.7 0.7 1" mesh="forearm" />
            <body name="wrist_1_link" pos="0 0 0.39225" quat="0.707107 0 0.707107 0">
                <inertial pos="0 0 0" quat="0.5 0.5 -0.5 0.5" mass="1.219" diaginertia="0.21942 0.111173 0.111173" />
                <geom type="mesh" rgba="0.7 0.7 0.7 1" mesh="wrist1" />
                <body name="wrist_2_link" pos="0 0.093 0">
                    <inertial pos="0 0 0" quat="0.5 0.5 -0.5 0.5" mass="1.219" diaginertia="0.21942 0.111173 0.111173" />
                    <geom type="mesh" rgba="0.7 0.7 0.7 1" mesh="wrist2" />
                    <body name="wrist_3_link" pos="0 0 0.09465">
                        <inertial pos="0 0 0" quat="0.5 0.5 -0.5 0.5"  mass="0.1879" diaginertia="0.033822 0.0171365 0.0171365" />
                        <geom type="mesh" rgba="0.7 0.7 0.7 1" friction="0.8 0.8 0.8" mesh="wrist3" />
                    </body>
                </body>
            </body>
        </body>
    </body>
</body>

Then I type the absolute direction but it gave me result that the file didn't exit. So how can I write the directory for the xml file?
Thanks again
Gloria

**Comment by rpapallas** [comment]
Hi Gloria,

Your xml file needs to be of the following structure: https://github.com/roboticsleeds/mujoco_ur5_model/blob/master/include/robot.xml

Just copy paste this XML file (you can exclude the mujoco include element)

**Comment by Gloriabhsfer** [comment]
> Hi Gloria,
> 
> Your xml file needs to be of the following structure: https://github.com/roboticsleeds/mujoco_ur5_model/blob/master/include/robot.xml
> 
Yes, I have tried copy this xml file and write the direction. But it gave me error like this:


> Please type the path to the xml file to be updated: '/home/gloria/mujoco_ur5_model/scripts/robot.xml' 
> Traceback (most recent call last):
>   File "set_ur5_joints.py", line 161, in <module>
>     update_xml_with_desired_joint_values(model_file_path, ur5)
>   File "set_ur5_joints.py", line 116, in update_xml_with_desired_joint_values
>     xml_document = minidom.parse(file_path)
>   File "/home/gloria/anaconda3/lib/python3.7/xml/dom/minidom.py", line 1958, in parse
>     return expatbuilder.parse(file)
>   File "/home/gloria/anaconda3/lib/python3.7/xml/dom/expatbuilder.py", line 910, in parse
>     with open(file, 'rb') as fp:
> FileNotFoundError: [Errno 2] No such file or directory: "'/home/gloria/mujoco_ur5_model/scripts/robot.xml' "

And I have tried put this xml file in script and type the name of it, there will be no error report but the simulator didn't change the configuration of UR5. So in your environment you can change the configuration of robot through xml file and quaternion python file right?
Thank you
Gloria

**Comment by rpapallas** [comment]
You are correct. It looks like you are using quotes (“) when you provide the path to the script, dont wrap the path in quotes.

Instead of:
'/home/gloria/mujoco_ur5_model/scripts/robot.xml'

Try:
/home/gloria/mujoco_ur5_model/scripts/robot.xml

Note that the script is updating the XML file and you get a static configuration of the robot every time.

**Comment by Gloriabhsfer** [comment]
Hi Rpapallas:
Thanks for your reply.
I find things works. Because I opened your example.xml file and it writes the robot XML file in the includes directory. So every time I modify the XML file in adjust directory and reopen the example.xml file in mujoco.  Then I can see the change in robot configuration. Thanks again for your good model and help.
Also, I find the ur5 model in mujoco website contained how to change robot joint [http://www.mujoco.org/forum/index.php?resources/universal-robots-ur5-robotiq-s-model-3-finger-gripper.22/](url)  Hope it can helps you.
As my question has been solved, I will close this issue. Thank you.
Gloria
